auto-mode: Skill activation syntax causes parameter validation failure, triggering stuck loop

## Problem

Auto-mode enters infinite dispatch loop when skill activation block uses function-call syntax `Skill('name')` which causes the LLM to pass `{name: "..."}` to the Skill tool instead of the required `{skill: "..."}` parameter. This triggers tool validation failures that are not properly recovered from, causing units to be re-dispatched repeatedly.

## Root Cause

**File:** `auto-prompts.js:335`

**Problematic code:**
```javascript
function formatSkillActivationBlock(skillNames) {
    if (skillNames.length === 0)
        return "";
    const calls = skillNames.map(name => `Call Skill('${name}')`).join('. ');
    return `<skill_activation>${calls}.</skill_activation>`;
}
```

The activation syntax `Call Skill('debug-like-expert')` appears to be a function call with a positional argument. The LLM interprets this as passing a parameter named `name`, but the Skill tool schema requires a parameter named `skill`:

```json
{
  "name": "Skill",
  "parameters": {
    "required": ["skill"],
    "properties": {
      "skill": { "type": "string" }
    }
  }
}
```

**Error observed:**
```
Validation failed for tool "Skill":
- skill: must have required property 'skill'
Received arguments: { "name": "debug-like-expert" }
```

## Expected Behavior

The skill activation syntax should explicitly show the parameter name to avoid LLM confusion. Change from:

```javascript
const calls = skillNames.map(name => `Call Skill('${name}')`).join('. ');
```

To:

```javascript
const calls = skillNames.map(name => `Call Skill({ skill: '${name}' })`).join('. ');
```

This makes the parameter name explicit and matches the tool schema.

## Environment

- GSD version: 2.42.0
- Model: qwen3.5-plus
- Unit: plan/slice-M003-S07

## Reproduction Context

User was executing S07 plan (JNI Symbol Fix for Android Dioxus framework). The skill activation block in the system prompt instructed:

```
<skill_activation>Call Skill('accessibility'). Call Skill('agent-browser'). Call Skill('debug-like-expert')...</skill_activation>
```

The LLM attempted to call `Skill('debug-like-expert')` but passed `{name: "debug-like-expert"}` instead of `{skill: "debug-like-expert"}`.

## Forensic Evidence

**Anomalies:**
- `[ERROR] stuck-loop`: Unit complete-milestone/M002-dbrk2n dispatched 5 times
- `[WARNING] stuck-loop`: Multiple units dispatched 2+ times (complete-slice, reassess-roadmap, research-slice)
- `[WARNING] error-trace`: Skill tool validation failure with wrong parameter name

**Error trace:**
```
Validation failed for tool "Skill":
- skill: must have required property 'skill'
Received arguments: {
  "name": "debug-like-expert"
}
```

**Impact:**
- Units fail verification due to tool call errors
- Recovery logic re-dispatches the same unit
- Infinite loop until manual intervention

---
*Auto-generated by `/gsd forensics`*

--- a/src/resources/extensions/gsd/auto-prompts.ts
+++ b/src/resources/extensions/gsd/auto-prompts.ts
@@ -419,12 +419,17 @@ function resolvePreferredSkillNames(
     .map(skill => normalizeSkillReference(skill.name));
 }
 
+/** Skill names must be lowercase alphanumeric with hyphens — reject anything else
+ *  to prevent prompt injection via crafted directory names. */
+const SAFE_SKILL_NAME = /^[a-z0-9][a-z0-9-]*$/;
+
 function formatSkillActivationBlock(skillNames: string[]): string {
-  if (skillNames.length === 0) return "";
+  const safe = skillNames.filter(name => SAFE_SKILL_NAME.test(name));
+  if (safe.length === 0) return "";
   // Use explicit parameter syntax so LLMs pass { skill: "..." } instead of { name: "..." }.
   // The function-call-like syntax `Skill('name')` led LLMs to infer a positional
   // parameter name, causing tool validation failures — see #2224.
-  const calls = skillNames.map(name => `Call Skill({ skill: '${name}' })`).join('. ');
+  const calls = safe.map(name => `Call Skill({ skill: '${name}' })`).join('. ');
   return `<skill_activation>${calls}.</skill_activation>`;
 }
 

--- a/src/resources/extensions/gsd/tests/skill-activation.test.ts
+++ b/src/resources/extensions/gsd/tests/skill-activation.test.ts
@@ -192,10 +192,11 @@ test("buildSkillActivationBlock does not activate skills from extraContext or ta
   }
 });
 
-test("buildSkillActivationBlock handles skill names with special characters safely", () => {
+test("buildSkillActivationBlock rejects skill names with special characters", () => {
   const base = makeTempBase();
   try {
-    // Skill names come from directory names — test that quotes/braces don't break the template
+    // Skill names with quotes, braces, or other non-alphanumeric characters are
+    // rejected by the SAFE_SKILL_NAME guard to prevent prompt injection.
     writeSkill(base, "my-skill's", "Skill with apostrophe in name.");
     loadOnlyTestSkills(base);
 
@@ -203,10 +204,29 @@ test("buildSkillActivationBlock handles skill names with special characters safe
       always_use_skills: ["my-skill's"],
     });
 
-    // The skill name is interpolated as-is — this documents current behavior.
-    // A future guard (e.g. /^[a-z0-9-]+$/) could reject such names.
+    // Unsafe skill name is filtered out — empty result
+    assert.equal(result, "");
+  } finally {
+    cleanup(base);
+  }
+});
+
+test("buildSkillActivationBlock allows valid skill names and rejects invalid ones", () => {
+  const base = makeTempBase();
+  try {
+    writeSkill(base, "react", "React skill.");
+    writeSkill(base, "bad'name", "Injection attempt.");
+    writeSkill(base, "good-skill-2", "Another valid skill.");
+    loadOnlyTestSkills(base);
+
+    const result = buildBlock(base, {}, {
+      always_use_skills: ["react", "bad'name", "good-skill-2"],
+    });
+
     assert.match(result, /skill_activation/);
-    assert.match(result, /my-skill's/);
+    assert.match(result, /Call Skill\(\{ skill: 'react' \}\)/);
+    assert.match(result, /Call Skill\(\{ skill: 'good-skill-2' \}\)/);
+    assert.doesNotMatch(result, /bad'name/);
   } finally {
     cleanup(base);
   }

